include/host/wasi/environ.h

```diff
@@ -38,6 +40,24 @@ class Environ {
 
   void fini() noexcept;
 
+  WasiExpect<void>
+  getAddrInfo(const char *Node, const char *Service,
+              const __wasi_addrinfo_t &Hint, uint32_t MaxResLength,
+              std::vector<struct __wasi_addrinfo_t *> *WasiAddrinfoArray,
+              std::vector<struct __wasi_sockaddr_t *> *WasiSockaddrArray,
+              std::vector<char *> *AiAddrSaDataArray,
+              std::vector<char *> *AiCanonnameArray,
```

Comment:
According to your implementation, all of these variable must not be `nullptr`, use reference instead.